src/main/java/com/example/producerconsumer/ProducerConsumerApplication.java

```diff
@@ -35,8 +37,13 @@ public void setNumberOfConsumers(int numberOfConsumers) {
     public static void main(String[] args) {
         SpringApplication.run(ProducerConsumerApplication.class, args);
 
+        //Tutaj raczej bym zrobił jakiś "runner" który byłby komponentem i się inicjował na @Init
+        //A tak już całkiem szczerze to bym springa tutaj nie angażował
+
```

Comment:
Ogólnie beany spring-we używałem w miejscach w  których wiedziałem że mogę/chcę użyć singleton-ów. Wie naturalnym wydawało mi się użycie serwisów. Mógłbym użyć klas statycznych albo własncy singletownów ale są łatwiejsze do utrzymania moim zdaniem.

Reply:
jak juz skomentowałem niżej - patrze na zależności - im mniej i słabsze tym lepiej. To się dotyczy metod, klas czy także bibliotek